source/DSCResources/DSC_CMSystemDiscovery/DSC_CMSystemDiscovery.psm1

```diff
@@ -0,0 +1,598 @@
+
+    .PARAMETER ScheduleCount
+        Specifies how often the recur interval is run.
+#>
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [Boolean]
+        $EnableDeltaDiscovery,
+
+        [Parameter()]
+        [ValidateRange(1,60)]
+        [UInt32]
+        $DeltaDiscoveryMins,
+
+        [Parameter()]
+        [Boolean]
+        $EnableFilteringExpiredLogon,
+
+        [Parameter()]
+        [UInt32]
+        $TimeSinceLastLogonDays,
+
+        [Parameter()]
+        [Boolean]
+        $EnableFilteringExpiredPassword,
+
+        [Parameter()]
+        [UInt32]
+        $TimeSinceLastPasswordUpdateDays,
+
+        [Parameter()]
+        [String[]]
+        $ADContainers,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToInclude,
+
+        [Parameter()]
+        [String[]]
+        $ADContainersToExclude,
+
+        [Parameter()]
+        [ValidateSet('None','Days','Hours','Minutes')]
+        [String]
+        $ScheduleInterval,
+
+        [Parameter()]
+        [UInt32]
+        $ScheduleCount
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -Enabled $Enabled
+    $result = $true
+
+    if ($Enabled -eq $true)
+    {
+        $testParams = @{
+            CurrentValues = $state
+            DesiredValues = $PSBoundParameters
+            ValuesToCheck = @('Enabled','EnableDeltaDiscovery','DeltaDiscoveryMins','EnableFilteringExpiredLogon',
+                              'TimeSinceLastLogonDays','EnableFilteringExpiredPassword','TimeSinceLastPasswordUpdateDays')
+        }
+
+        $result = Test-DscParameterState @testParams -TurnOffTypeChecking -Verbose
+
+        if ($PSBoundParameters.ContainsKey('ScheduleInterval'))
+        {
+            if ($ScheduleInterval -ne 'None' -and -not $PSBoundParameters.ContainsKey('ScheduleCount'))
+            {
+                Write-Verbose -Message $script:localizedData.IntervalCount
+                $result = $false
+            }
+            else
+            {
+                if ($ScheduleInterval -ne $state.SCheduleInterval)
+                {
+                    Write-Verbose -Message ($script:localizedData.SIntervalTest -f $ScheduleInterval, $State.ScheduleInterval)
+                    $result = $false
+                }
+
+                if (($ScheduleInterval -ne 'None') -and ($ScheduleCount -ne $state.ScheduleCount))
+                {
+                    Write-Verbose -Message ($script:localizedData.SCountTest -f $ScheduleCount, $State.ScheduleCount)
+                    $result = $false
+                }
+            }
+        }
+
+        if (($ADContainers) -or ($ADContainersToInclude))
```

Comment:
Should do a write warning here that if ADContainers are specified alongside ADContainersToInclude or ADContainersToExclude, that the incl. and excl. params will be ignored so that the user knows.